src/components/IndexTable/IndexTable.tsx

```diff
@@ -0,0 +1,619 @@
+  sort?: React.ReactNode;
+  onMoreActionPopoverToggle?(isOpen: boolean): void;
+}
+
+export interface TableHeadingRect {
+  offsetWidth: number;
+  offsetLeft: number;
+}
+
+const SCROLL_BAR_PADDING = 4;
+
+export function IndexTable({
+  headings,
+  bulkActions = [],
+  promotedBulkActions = [],
+  children,
+  emptyState,
+  onMoreActionPopoverToggle,
+  sort,
+}: IndexTableProps) {
+  const {
+    loading,
+    bulkSelectState,
+    resourceName,
+    bulkActionsAccessibilityLabel,
+    selectMode,
+    paginatedSelectAllText,
+    itemCount,
+    hasMoreItems,
+    selectedItemsCount,
+    condensed,
+  } = useIndexValue();
+  const handleSelectionChange = useIndexSelectionChange();
+  const {newDesignLanguage} = useFeatures();
+  const i18n = useI18n();
+
+  const {
+    value: hasMoreLeftColumns,
+    toggle: toggleHasMoreLeftColumns,
+  } = useToggle(false);
+  const {
+    value: hasMoreRightColumns,
+    toggle: toggleHasMoreRightColumns,
+  } = useToggle(false);
+
+  const onboardingScrollButtons = useRef(false);
+  const tablePosition = useRef({top: 0, left: 0});
+  const tableHeadingRects = useRef<TableHeadingRect[]>([]);
+
+  const scrollableContainerElement = useRef<HTMLDivElement>(null);
+  const tableElement = useRef<HTMLTableElement>(null);
+  const [tableInitialized, setTableInitialized] = useState(false);
+
+  const [isSmallScreenSelectable, setIsSmallScreenSelectable] = useState(false);
+
+  const stickyHeaderElement = useRef<HTMLDivElement>(null);
+  const scrollBarElement = useRef<HTMLDivElement>(null);
+  const scrollingWithBar = useRef(false);
+  const scrollingContainer = useRef(false);
+
+  const tableBodyRef = useCallback(
+    (node) => {
+      if (node !== null && !tableInitialized) {
+        setTableInitialized(true);
+      }
+    },
+    [tableInitialized],
+  );
+
+  const toggleIsSmallScreenSelectable = useCallback(() => {
+    setIsSmallScreenSelectable((value) => !value);
+  }, []);
+
+  const handleSelectAllItemsInStore = useCallback(() => {
+    handleSelectionChange(
+      selectedItemsCount === SELECT_ALL_ITEMS
+        ? SelectionType.Page
+        : SelectionType.All,
+      true,
+    );
+  }, [handleSelectionChange, selectedItemsCount]);
+
+  const resizeTableHeadings = useCallback(() => {
+    if (!tableElement.current || !scrollableContainerElement.current) {
+      return;
+    }
+
+    const measuredTableHeadingRects = Array.from(
+      tableElement.current.querySelectorAll('[data-index-table-heading]'),
+    ).map((heading) => ({
+      offsetWidth: heading instanceof HTMLElement ? heading.offsetWidth : 0,
+      offsetLeft: heading instanceof HTMLElement ? heading.offsetLeft : 0,
+    }));
+    const boundingRect = scrollableContainerElement.current.getBoundingClientRect();
+    tablePosition.current = {top: boundingRect.top, left: boundingRect.left};
+    tableHeadingRects.current = measuredTableHeadingRects;
+  }, []);
+
+  // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
If you convert this to `useMemo` instead, you shouldn't need the eslint disable like we do [elsewhere](https://github.com/Shopify/polaris-react/blob/master/src/components/ActionMenu/components/Actions/Actions.tsx#L120).